django_migration_linter/migration_linter.py

```diff
@@ -61,51 +63,80 @@ def __init__(self, project_path, **kwargs):
         self.nb_total = 0
 
         # Initialise cache. Read from old, write to new to prune old entries.
-        if not self.no_cache:
-            self.old_cache = Cache(self.django_path, self.cache_path)
-            self.new_cache = Cache(self.django_path, self.cache_path)
+        if self.should_use_cache():
+            self.old_cache = Cache(self.django_path, self.database, self.cache_path)
+            self.new_cache = Cache(self.django_path, self.database, self.cache_path)
             self.old_cache.load()
 
+    def should_use_cache(self):
+        return self.django_path and not self.no_cache
+
+    def lint_all_migrations(self, git_commit_id=None):
+        # Collect migrations
+        if git_commit_id:
+            migrations = self._gather_migrations_git(git_commit_id)
+        else:
+            migrations = self._gather_all_migrations()
+
+        # Lint those migrations
+        sorted_migrations = sorted(
+            migrations, key=lambda migration: (migration.app_label, migration.name)
+        )
+        for m in sorted_migrations:
+            self.lint_migration(m)
+
+        if self.should_use_cache():
+            self.new_cache.save()
+
     def lint_migration(self, migration):
-        app_name = migration.app_name
+        app_label = migration.app_label
         migration_name = migration.name
-        print("({0}, {1})... ".format(app_name, migration_name), end="")
+        print("({0}, {1})... ".format(app_label, migration_name), end="")
         self.nb_total += 1
 
-        md5hash = migration.get_md5hash()
-        if not self.no_cache and md5hash in self.old_cache:
-            self.lint_cached_migration(md5hash)
-            return
+        md5hash = self.get_migration_hash(app_label, migration_name)
 
-        if self.should_ignore_migration(app_name, migration_name):
+        if self.should_ignore_migration(app_label, migration_name):
```

Comment:
as mentioned in https://github.com/3YOURMIND/django-migration-linter/issues/57#issuecomment-491286030 i would suggest to provide the `migration` to this method and make it check, if has an operation that `isinstanceof` `IgnoreMigration`. This would skip the later time consuming `get_sql` and `analyse_sql_statements` steps.